docs/exit_validation.md

```diff
@@ -1,48 +1,74 @@
+3. If an exit is invalid and unchallenged within a short period from it's finalization it must be challenged **AND** watcher must not allow spending and depositing
+    - because otherwise, we are "tightly" following the minimum safety requirements
+    - because otherwise, unchallenged exits may break the chain without honest holders having time to escape.
+    This is the case when an exit, a spend invalidating it and exit of that coincide and are unchallenged.
+4. (optional) If a finalization of an invalid exit occurs, watcher must prompt an exit without guarantees of recovery of funds
+
+Conditions 3 and 4 can be represented jointly, by **continuous** (every child block) validation of the exits.
+If any exit is invalid and its finalization is near, then actions listed under 3 and 4 should take place
+
+All that takes as an assumption that:
+  - the user's funds must be safe even if the user syncs and validates the chain periodically (but not less frequently than required)
+  - user needs to have the ability to constantly spend their UTXOs, thereby requiring more stringent validity checking of exits
+
+### Example cases for above actions:
+1. The exit is processed after the tx, so it is known to be invalid from the start. Should emit an `:invalid_exit` event.
+2. The exit is processed before the tx, but the tx is within `sla_margin` period, so the child chain is valid.
+Causes an `:invalid_exit` event.
+3. The exit is processed before the tx, but tx is after `sla_margin` period, so the child chain is byzantine.
+Causes to emit an `:unchallenged_exit` event.
+3. An exit is unchallenged within `sla_margin` period, so the child chain is jeopardized.
+Causes to emit an `:unchallenged_exit` event.
+4. (optional) Continuation of cases (1) or (2) where the exit never gets challenged and finalizes.
+Causes to emit an `:invalid_finalization` event
+
+### Solution for above cases:
+2. `ExitProcessor` periodically pulls open exit requests from root chain contract
 3. For every open exit request run `State.utxo_exists?` method
-    * if `true` -> noop, 
+    * if `true` -> noop,
     * if `false` -> emit `:invalid_exit` event  which leads to challenge
-    * if `false` and there is less than `SLA` time till finalization -> prompt exit
-4. Spend utxos in `State` only on `:finalize_exits` event
+    * if `false` and there is less than `sla_margin` time till finalization -> `:unchallenged_exit`
+4. Spend utxos in `State` on exit finalization or challenging
+
+### Mandatory automatic challenging with a strategy + spend prohibition
+
+The above notifications aren't enough to guarantee soundness of the chain.
+
+By the power of defaults we should:
+  - force the user running the Watcher to keep a (modestly) funded Ethereum account (`challenger`), that will be used to challenge **automatically** all of the invalid exits seen
+  - do the automatic challenging according to some randomized strategy, that will ensure that the challenges won't burn each other
```

Comment:
So the `watcher` will use/keep user private key to make those automatic challenges to the root chain contract